docs/pages/auth-concept/authentication.md

```diff
@@ -0,0 +1,181 @@
+---
+title: 'Authentication functions'
+weight: 3
+---
+
+# Getting started with Authentication functions
+
+Authentication in Glee can be done using authentication functions. Authentication functions are files that export either one or both of the `clientAuth` and `serverAuth` Node.js functions:
+
+```js
+/* websocket.js */
+
+export async function serverAuth({ authProps, done }) {
+  //server auth logic
+}
+
+export async function clientAuth({ parsedAsyncAPI, serverName }) {
+  //client auth logic
+}
+```
+
+Glee looks for authentication files in the `auth` directory by default but it can be configured using [glee config file](../config-file.md).
+The name of the authentication file should be the name of the targeted server that the authentication logic should work for.
+
+## Supported Authentication Values in asyncapi.yaml file
+
+AsyncAPI currently supports a variety of authentication formats as specified in the [documentation](https://www.asyncapi.com/docs/reference/specification/v2.6.0#securitySchemeObject), however Glee supports the following authentication schemas.
+
+- userPassword
+- Bearer token
+- HttpPApiKey
+- ApiKey
+
+A sample `asyncapi.yaml` for a server with security requirements and a `userPassword` security schemes is shown below:
+
+```yaml
+##server asyncAPI schema
+asyncapi: 2.6.0
+info:
+  title: AsyncAPI IMDB server
+  version: 1.0.0
+  description: This app is a dummy server that would stream the trending/upcoming anime.
+servers:
+  trendingAnimeServer:
+    url: 'http://localhost:8081'
+    protocol: http
+    security:
+      - userPass: []
+
+  ...
+
+components:
+  securitySchemes:
+    userPass:
+      type: userPassword
+
+```
+
+A sample `asyncapi.yaml` for a client that implements some of the requirements of the server above:
+
+```yaml
+##client asyncAPI schema
+servers:
+  trendingAnime:
+    url: http://localhost:8081
+    protocol: http
+    security:
+      - userPass: []
+  testwebhook:
+    url: ws://localhost:9000
+    protocol: ws
+x-remoteServers:
+  - trendingAnime
+
+  ...
+
+components:
+  securitySchemes:
+    userPass:
+      type: userPassword
+
+```
+
+**The Client asyncapi.yaml file does not need to implement all the security requirements in the server, it only needs to implement the ones that it uses.**
```

Comment:
an example would be great here.

Reply:
Are you talking abt `securitySchemas` instances that client uses `userPass` here only excluding other requirements like token or httpApikey etc.?

Reply:
TBH, I don't know what this line means. 😆 that's why I was asking for an example.
cc: @Souvikns @oviecodes

Reply:
> Client asyncapi.yaml file only needs to contain the security requirements

Sorry @oviecodes, do you mean client uses `userPass` here only excluding others?

Reply:
@afzal442 since almost always there is only one way for a client to authenticate against a server, I suggest we remove this line. It causes confusion.

Reply:
instead expanded it like now